/**
 * Copyright 2018-2021 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

syntax = "proto3";

// Constants for our two "RootContainers" - the top-level containers
// that `st.foo` and `st.sidebar.foo` write to. Each value corresponds
// to the first index in any `delta_path` that uses these containers.
// (This enum is not used in any protobuf messages; we declare it here so
// that the values remain synced between Python and the frontend.)
enum RootContainer {
  MAIN = 0;
  SIDEBAR = 1;
}
